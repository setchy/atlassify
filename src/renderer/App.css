html,
body,
#root {
  height: 100%;
  -webkit-user-select: none;
}

html::-webkit-scrollbar,
div.flex-grow::-webkit-scrollbar {
  width: 10px;
}

html::-webkit-scrollbar-thumb,
div.flex-grow::-webkit-scrollbar-thumb {
  background-color: #c1c1c1;
  border-radius: 10px;
}

html::-webkit-scrollbar-thumb:hover,
div.flex-grow::-webkit-scrollbar-thumb:hover {
  background-color: #a8a8a8;
  border-radius: 10px;
}

body {
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  margin: 0;
  cursor: default;
}

/**
 * Set emoji size according to surrounding text.
 * Ref: https://github.com/jdecked/twemoji?tab=readme-ov-file#inline-styles
 */
img.emoji {
  height: 1em;
  width: 1em;
  margin: 0 0.05em 0 0.1em;
  vertical-align: -0.1em;
}
