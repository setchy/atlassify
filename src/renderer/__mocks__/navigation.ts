const mockNavigate = vi.fn();

vi.mock('react-router-dom', async () => ({
  ...(await vi.importActual('react-router-dom')),
  useNavigate: () => mockNavigate,
}));

export { mockNavigate };
