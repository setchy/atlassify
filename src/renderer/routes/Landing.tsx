import { type FC, useContext, useEffect } from 'react';
import { useNavigate } from 'react-router-dom';

import Button from '@atlaskit/button/new';
import Heading from '@atlaskit/heading';
import {
  AtlasIcon,
  AtlassianIcon,
  BitbucketIcon,
  CompassIcon,
  ConfluenceIcon,
  JiraIcon,
  JiraProductDiscoveryIcon,
  JiraServiceManagementIcon,
  TrelloIcon,
} from '@atlaskit/logo';
import { Inline, Stack, Text } from '@atlaskit/primitives';
import Tooltip from '@atlaskit/tooltip';

import { Centered } from '../components/primitives/Centered';
import { AppContext } from '../context/App';
import { showWindow } from '../utils/comms';

export const LandingRoute: FC = () => {
  const navigate = useNavigate();
  const { isLoggedIn } = useContext(AppContext);

  useEffect(() => {
    if (isLoggedIn) {
      showWindow();
      navigate('/', { replace: true });
    }
  }, [isLoggedIn]);

  return (
    <Centered>
      <Stack alignBlock="center" alignInline="center" space="space.200">
        <AtlasIcon appearance="brand" size="xlarge" />
        <Stack alignInline="center">
          <Heading size="large">Atlassian notifications</Heading>
          <Text size="large">on your menu bar</Text>
        </Stack>
        <Inline space="space.100">
          <BitbucketIcon size="small" appearance="neutral" />
          <CompassIcon size="small" appearance="neutral" />
          <ConfluenceIcon size="small" appearance="neutral" />
          <JiraIcon size="small" appearance="neutral" />
          <JiraProductDiscoveryIcon size="small" appearance="neutral" />
          <JiraServiceManagementIcon size="small" appearance="neutral" />
          <TrelloIcon size="small" appearance="neutral" />
        </Inline>
        <Tooltip content="Login with Atlassian">
          <Button
            appearance="primary"
            spacing="default"
            iconBefore={(iconProps) => (
              <AtlassianIcon {...iconProps} size="small" />
            )}
            onClick={() => navigate('/login')}
            testId="login"
          >
            Login
          </Button>
        </Tooltip>
      </Stack>
    </Centered>
  );
};
