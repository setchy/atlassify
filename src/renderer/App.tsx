import { useContext } from 'react';
import {
  Navigate,
  Route,
  HashRouter as Router,
  Routes,
  useLocation,
} from 'react-router-dom';

import { AppContext, AppProvider } from './context/App';
import { AccountsRoute } from './routes/Accounts';
import { FiltersRoute } from './routes/Filters';
import { LandingRoute } from './routes/Landing';
import { LoginRoute } from './routes/Login';
import { NotificationsRoute } from './routes/Notifications';
import { SettingsRoute } from './routes/Settings';

import './App.css';
import { AppLayout } from './components/layout/AppLayout';

function RequireAuth({ children }) {
  const { isLoggedIn } = useContext(AppContext);
  const location = useLocation();

  return isLoggedIn ? (
    children
  ) : (
    <Navigate to="/landing" replace state={{ from: location }} />
  );
}

export const App = () => {
  return (
    <AppProvider>
      <Router>
        <AppLayout>
          <Routes>
            <Route
              path="/"
              element={
                <RequireAuth>
                  <NotificationsRoute />
                </RequireAuth>
              }
            />
            <Route
              path="/filters"
              element={
                <RequireAuth>
                  <FiltersRoute />
                </RequireAuth>
              }
            />
            <Route
              path="/settings"
              element={
                <RequireAuth>
                  <SettingsRoute />
                </RequireAuth>
              }
            />
            <Route
              path="/accounts"
              element={
                <RequireAuth>
                  <AccountsRoute />
                </RequireAuth>
              }
            />
            <Route path="/landing" element={<LandingRoute />} />
            <Route path="/login" element={<LoginRoute />} />
          </Routes>
        </AppLayout>
      </Router>
    </AppProvider>
  );
};
