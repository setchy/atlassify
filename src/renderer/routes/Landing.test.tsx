import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import { mockNavigate } from '../__mocks__/navigation';

import { AppContext } from '../context/App';
import * as comms from '../utils/comms';
import { LandingRoute } from './Landing';

describe('renderer/routes/Landing.tsx', () => {
  afterEach(() => {
    vi.clearAllMocks();
  });

  it('should render itself & its children', () => {
    const tree = render(
      <MemoryRouter>
        <LandingRoute />
      </MemoryRouter>,
    );

    expect(tree).toMatchSnapshot();
  });

  it('should redirect to notifications once logged in', () => {
    const showWindowMock = vi.spyOn(comms, 'showWindow');

    const { rerender } = render(
      <AppContext.Provider value={{ isLoggedIn: false }}>
        <MemoryRouter>
          <LandingRoute />
        </MemoryRouter>
      </AppContext.Provider>,
    );

    rerender(
      <AppContext.Provider value={{ isLoggedIn: true }}>
        <MemoryRouter>
          <LandingRoute />
        </MemoryRouter>
      </AppContext.Provider>,
    );

    expect(showWindowMock).toHaveBeenCalledTimes(1);
    expect(mockNavigate).toHaveBeenNthCalledWith(1, '/', { replace: true });
  });

  it('should navigate to login with api token', () => {
    render(
      <MemoryRouter>
        <LandingRoute />
      </MemoryRouter>,
    );

    fireEvent.click(screen.getByTestId('login'));

    expect(mockNavigate).toHaveBeenNthCalledWith(1, '/login');
  });
});
