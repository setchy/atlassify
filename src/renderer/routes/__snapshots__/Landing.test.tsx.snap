// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer/routes/Landing.tsx should render itself & its children 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-1 flex-col items-center justify-center"
      >
        <div
          class="css-11aluc9"
        >
          <span
            aria-label="Atlas"
            class="css-1eivgv0"
            role="img"
          >
            <svg
              aria-hidden="true"
              fill="none"
              focusable="false"
              height="32"
              viewBox="0 0 32 32"
              xmlns="http://www.w3.org/2000/svg"
            >
              
    
              <g
                fill="#1868DB"
              >
                
      
                <path
                  d="M16 28.532c-6.656 0-12.068-5.412-12.068-12.068S9.344 4.396 16 4.396v1.921c-5.597 0-10.146 4.549-10.146 10.146S10.403 26.609 16 26.609zm0 0v-1.921c5.597 0 10.146-4.549 10.146-10.146h1.921c0 6.656-5.412 12.068-12.068 12.068zm12.068-12.068h-1.921c0-5.597-4.549-10.146-10.146-10.146V4.397c6.656 0 12.068 5.412 12.068 12.068zm-6.928 3.695c-1.28-.251-2.292-.706-3.973-1.467-.464-.204-.974-.436-1.559-.696a59 59 0 0 0-1.003-.436c-2.534-1.123-3.926-1.736-5.634-1.736-1.198 0-2.098.325-2.6.937a1.72 1.72 0 0 0-.38 1.448l-1.885.362a3.67 3.67 0 0 1 .78-3.035c.613-.743 1.82-1.634 4.085-1.634 2.116 0 3.824.761 6.414 1.903.316.14.641.288.994.436.594.26 1.114.492 1.578.706 1.606.724 2.488 1.123 3.555 1.327l-.371 1.885Zm1.877.185c-.64 0-1.253-.056-1.875-.185l.371-1.885c.502.102.994.149 1.504.149 1.188 0 2.089-.298 2.609-.864s.539-1.253.51-1.615l1.913-.167c.102 1.169-.251 2.265-1.003 3.082-.622.677-1.82 1.485-4.029 1.485M16 28.532v-1.921c2.757 0 4.4-2.813 4.4-7.52s-1.67-8.652-5.096-12.3l1.402-1.318c3.778 4.02 5.616 8.476 5.616 13.618 0 5.821-2.423 9.441-6.322 9.441m-.001 0c-4.15 0-6.526-3.379-6.526-9.273 0-6.879 3.008-10.722 5.82-13.776l1.412 1.299c-3.277 3.565-5.31 6.888-5.31 12.476 0 6.08 2.506 7.352 4.604 7.352z"
                />
                
      
                <path
                  d="M10.857 17.467a2.247 2.247 0 1 0 0-4.494 2.247 2.247 0 0 0 0 4.494m10.385 3.871a2.247 2.247 0 1 0 0-4.494 2.247 2.247 0 0 0 0 4.494M15.998 7.96a2.247 2.247 0 1 0 0-4.493 2.247 2.247 0 0 0 0 4.494Z"
                />
                
    
              </g>
              
  
            </svg>
          </span>
          <div
            class="css-1ezbfq2"
          >
            <h2
              class="css-1i4hnz3"
            >
              Atlassian notifications
            </h2>
            <span
              class="css-11ssxzq"
            >
              on your menu bar
            </span>
          </div>
          <div
            class="css-gokkc4"
          >
            <span
              aria-label="Bitbucket"
              class="css-1eu9lrx"
              role="img"
              style="--logo-color: inherit; --logo-fill: currentColor;"
            >
              <svg
                aria-hidden="true"
                fill="none"
                focusable="false"
                height="32"
                viewBox="0 0 32 32"
                xmlns="http://www.w3.org/2000/svg"
              >
                
      
                <path
                  d="m25.586 15.819-1.618 9.885c-.106.598-.528.95-1.126.95H9.122c-.598 0-1.02-.352-1.126-.95L5.146 8.08c-.105-.598.212-.985.775-.985h20.123c.562 0 .88.387.773.985l-.773 4.644c-.106.668-.493.95-1.126.95H12.816c-.176 0-.282.105-.246.316l.95 5.84c.035.14.14.246.281.246h4.362c.141 0 .246-.105.282-.246l.668-4.222c.07-.527.422-.738.915-.738h4.75c.703 0 .914.351.808.95"
                  fill="#6C6F77"
                />
                
    
              </svg>
            </span>
            <span
              aria-label="Compass"
              class="css-1eu9lrx"
              role="img"
              style="--logo-color: inherit; --logo-fill: currentColor;"
            >
              <svg
                aria-hidden="true"
                fill="none"
                focusable="false"
                height="32"
                viewBox="0 0 32 32"
                xmlns="http://www.w3.org/2000/svg"
              >
                
      
                <path
                  d="M15.667 15.66v5.942H9.713V15.66zV9.722H4.807c-.58.004-1.047.47-1.043 1.04v15.755a1.04 1.04 0 0 0 1.028 1.054h15.771a1.04 1.04 0 0 0 1.042-1.04V15.66zm0-11.88v5.942h5.967v5.938h5.935V4.82a1.04 1.04 0 0 0-1.028-1.053h-.014z"
                  fill="#6C6F77"
                />
                
      
              </svg>
              
    
            </span>
            <span
              aria-label="Confluence"
              class="css-1eu9lrx"
              role="img"
              style="--logo-color: inherit; --logo-fill: currentColor;"
            >
              <svg
                aria-hidden="true"
                fill="none"
                focusable="false"
                height="32"
                viewBox="0 0 32 32"
                xmlns="http://www.w3.org/2000/svg"
              >
                
      
                <path
                  d="M25.769 20.234c-6.584-3.183-8.507-3.66-11.282-3.66-3.254 0-6.028 1.355-8.507 5.16l-.406.622c-.333.513-.407.696-.407.915s.11.403.517.659l4.18 2.598c.222.146.407.22.592.22.222 0 .37-.11.592-.44l.665-1.024c1.036-1.574 1.96-2.086 3.144-2.086 1.036 0 2.257.293 3.773 1.025l4.365 2.049c.443.22.924.11 1.146-.403l2.071-4.537c.222-.512.074-.842-.444-1.098M6.572 12.22c6.584 3.184 8.507 3.66 11.281 3.66 3.255 0 6.03-1.354 8.507-5.16l.407-.622c.333-.512.407-.695.407-.915s-.11-.402-.518-.658l-4.18-2.598c-.221-.147-.406-.22-.591-.22-.222 0-.37.11-.592.44l-.666 1.024c-1.035 1.573-1.96 2.086-3.144 2.086-1.035 0-2.256-.293-3.772-1.025L9.346 6.183c-.444-.22-.924-.11-1.146.402l-2.072 4.538c-.222.512-.074.841.444 1.098"
                  fill="#6C6F77"
                />
                
    
              </svg>
            </span>
            <span
              aria-label="Jira"
              class="css-1eu9lrx"
              role="img"
              style="--logo-color: inherit; --logo-fill: currentColor;"
            >
              <svg
                aria-hidden="true"
                fill="none"
                focusable="false"
                height="32"
                viewBox="0 0 32 32"
                xmlns="http://www.w3.org/2000/svg"
              >
                
      
                <path
                  d="M11.034 21.99h-2.22c-3.346 0-5.747-2.05-5.747-5.052h11.932c.619 0 1.019.44 1.019 1.062v12.007c-2.983 0-4.984-2.416-4.984-5.784zm5.893-5.967h-2.219c-3.347 0-5.748-2.013-5.748-5.015h11.933c.618 0 1.055.402 1.055 1.025V24.04c-2.983 0-5.02-2.416-5.02-5.784zm5.93-5.93h-2.219c-3.347 0-5.748-2.05-5.748-5.052h11.933c.618 0 1.018.439 1.018 1.025v12.007c-2.983 0-4.984-2.416-4.984-5.784z"
                  fill="#6C6F77"
                />
                
    
              </svg>
            </span>
            <span
              aria-label="Jira Product Discovery"
              class="css-1eu9lrx"
              role="img"
              style="--logo-color: inherit; --logo-fill: currentColor;"
            >
              <svg
                aria-hidden="true"
                fill="none"
                focusable="false"
                height="32"
                viewBox="0 0 32 32"
                xmlns="http://www.w3.org/2000/svg"
              >
                
      
                <path
                  d="M17.655 5.884a15.3 15.3 0 0 0 8.75 0c.257-.056.428.114.37.37-.826 2.985-.826 5.742 0 8.727.058.228-.113.427-.37.342a16.33 16.33 0 0 0-8.75 0c-.228.085-.428-.114-.342-.342.77-2.985.77-5.742 0-8.727-.086-.256.114-.426.342-.37m1.169 10.576v.483c0 2.246-1.568 3.894-3.506 3.894a3.474 3.474 0 0 1-3.477-3.496c0-1.933 1.567-3.468 3.762-3.468h.57V7.078c-.285-.028-.57-.028-.855-.028C9.618 7.05 5 11.655 5 17.37c0 5.713 4.617 10.29 10.318 10.29s10.346-4.605 10.346-10.29v-.91z"
                  fill="#6C6F77"
                />
                
    
              </svg>
            </span>
            <span
              aria-label="Jira Service Management"
              class="css-1eu9lrx"
              role="img"
              style="--logo-color: inherit; --logo-fill: currentColor;"
            >
              <svg
                aria-hidden="true"
                fill="none"
                focusable="false"
                height="32"
                viewBox="0 0 32 32"
                xmlns="http://www.w3.org/2000/svg"
              >
                
      
                <path
                  d="M18.79 13.257h7.325c1.1 0 1.476 1.046.805 1.878L15.464 29.274c-3.702-2.951-3.353-7.62-.644-11.027zm-5.66 5.634H5.806c-1.1 0-1.476-1.046-.805-1.878L16.457 2.874c3.702 2.951 3.3 7.566.617 11z"
                  fill="#6C6F77"
                />
                
    
              </svg>
            </span>
            <span
              aria-label="Trello"
              class="css-1eu9lrx"
              role="img"
              style="--logo-color: inherit; --logo-fill: currentColor;"
            >
              <svg
                aria-hidden="true"
                fill="none"
                focusable="false"
                height="32"
                viewBox="0 0 32 32"
                xmlns="http://www.w3.org/2000/svg"
              >
                
      
                <path
                  clip-rule="evenodd"
                  d="M24.286 5.337H7.714a2.585 2.585 0 0 0-2.585 2.585v16.572a2.585 2.585 0 0 0 2.585 2.584h16.572a2.585 2.585 0 0 0 2.585-2.584V7.922a2.585 2.585 0 0 0-2.585-2.585m-9.791 15.66a.86.86 0 0 1-.859.858h-3.634a.86.86 0 0 1-.859-.859V10.21a.86.86 0 0 1 .859-.86h3.634a.86.86 0 0 1 .859.86zm8.362-4.953a.86.86 0 0 1-.859.859h-3.634a.86.86 0 0 1-.859-.859V10.21a.86.86 0 0 1 .859-.86h3.634a.86.86 0 0 1 .859.86z"
                  fill="#6C6F77"
                  fill-rule="evenodd"
                />
                
    
              </svg>
            </span>
          </div>
          <div
            role="presentation"
          >
            <button
              class="css-1ilq2uw"
              data-testid="login"
              type="button"
            >
              <span
                class="css-12rb8n9"
              >
                <span
                  class="css-1gskvga"
                  style="--logo-color: inherit; --logo-fill: currentColor;"
                >
                  <svg
                    aria-hidden="true"
                    fill="none"
                    focusable="false"
                    height="32"
                    viewBox="0 0 32 32"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    
      
                    <path
                      d="M27.545 24.378 16.96 3.208c-.208-.458-.417-.541-.667-.541-.208 0-.458.083-.708.5-1.5 2.375-2.167 5.125-2.167 8 0 4.001 2.042 7.752 5.042 13.795.334.666.584.791 1.167.791h7.335c.541 0 .833-.208.833-.625 0-.208-.042-.333-.25-.75M12.168 14.377c-.834-1.25-1.084-1.334-1.292-1.334s-.333.083-.708.834L4.875 24.46c-.167.334-.208.459-.208.625 0 .334.291.667.916.667h7.46c.5 0 .875-.416 1.083-1.208.25-1 .334-1.876.334-2.917 0-2.917-1.292-5.751-2.292-7.251"
                      fill="inherit"
                    />
                    
    
                  </svg>
                </span>
              </span>
              <span
                class="css-1nqby2s"
              >
                Login
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-1 flex-col items-center justify-center"
    >
      <div
        class="css-11aluc9"
      >
        <span
          aria-label="Atlas"
          class="css-1eivgv0"
          role="img"
        >
          <svg
            aria-hidden="true"
            fill="none"
            focusable="false"
            height="32"
            viewBox="0 0 32 32"
            xmlns="http://www.w3.org/2000/svg"
          >
            
    
            <g
              fill="#1868DB"
            >
              
      
              <path
                d="M16 28.532c-6.656 0-12.068-5.412-12.068-12.068S9.344 4.396 16 4.396v1.921c-5.597 0-10.146 4.549-10.146 10.146S10.403 26.609 16 26.609zm0 0v-1.921c5.597 0 10.146-4.549 10.146-10.146h1.921c0 6.656-5.412 12.068-12.068 12.068zm12.068-12.068h-1.921c0-5.597-4.549-10.146-10.146-10.146V4.397c6.656 0 12.068 5.412 12.068 12.068zm-6.928 3.695c-1.28-.251-2.292-.706-3.973-1.467-.464-.204-.974-.436-1.559-.696a59 59 0 0 0-1.003-.436c-2.534-1.123-3.926-1.736-5.634-1.736-1.198 0-2.098.325-2.6.937a1.72 1.72 0 0 0-.38 1.448l-1.885.362a3.67 3.67 0 0 1 .78-3.035c.613-.743 1.82-1.634 4.085-1.634 2.116 0 3.824.761 6.414 1.903.316.14.641.288.994.436.594.26 1.114.492 1.578.706 1.606.724 2.488 1.123 3.555 1.327l-.371 1.885Zm1.877.185c-.64 0-1.253-.056-1.875-.185l.371-1.885c.502.102.994.149 1.504.149 1.188 0 2.089-.298 2.609-.864s.539-1.253.51-1.615l1.913-.167c.102 1.169-.251 2.265-1.003 3.082-.622.677-1.82 1.485-4.029 1.485M16 28.532v-1.921c2.757 0 4.4-2.813 4.4-7.52s-1.67-8.652-5.096-12.3l1.402-1.318c3.778 4.02 5.616 8.476 5.616 13.618 0 5.821-2.423 9.441-6.322 9.441m-.001 0c-4.15 0-6.526-3.379-6.526-9.273 0-6.879 3.008-10.722 5.82-13.776l1.412 1.299c-3.277 3.565-5.31 6.888-5.31 12.476 0 6.08 2.506 7.352 4.604 7.352z"
              />
              
      
              <path
                d="M10.857 17.467a2.247 2.247 0 1 0 0-4.494 2.247 2.247 0 0 0 0 4.494m10.385 3.871a2.247 2.247 0 1 0 0-4.494 2.247 2.247 0 0 0 0 4.494M15.998 7.96a2.247 2.247 0 1 0 0-4.493 2.247 2.247 0 0 0 0 4.494Z"
              />
              
    
            </g>
            
  
          </svg>
        </span>
        <div
          class="css-1ezbfq2"
        >
          <h2
            class="css-1i4hnz3"
          >
            Atlassian notifications
          </h2>
          <span
            class="css-11ssxzq"
          >
            on your menu bar
          </span>
        </div>
        <div
          class="css-gokkc4"
        >
          <span
            aria-label="Bitbucket"
            class="css-1eu9lrx"
            role="img"
            style="--logo-color: inherit; --logo-fill: currentColor;"
          >
            <svg
              aria-hidden="true"
              fill="none"
              focusable="false"
              height="32"
              viewBox="0 0 32 32"
              xmlns="http://www.w3.org/2000/svg"
            >
              
      
              <path
                d="m25.586 15.819-1.618 9.885c-.106.598-.528.95-1.126.95H9.122c-.598 0-1.02-.352-1.126-.95L5.146 8.08c-.105-.598.212-.985.775-.985h20.123c.562 0 .88.387.773.985l-.773 4.644c-.106.668-.493.95-1.126.95H12.816c-.176 0-.282.105-.246.316l.95 5.84c.035.14.14.246.281.246h4.362c.141 0 .246-.105.282-.246l.668-4.222c.07-.527.422-.738.915-.738h4.75c.703 0 .914.351.808.95"
                fill="#6C6F77"
              />
              
    
            </svg>
          </span>
          <span
            aria-label="Compass"
            class="css-1eu9lrx"
            role="img"
            style="--logo-color: inherit; --logo-fill: currentColor;"
          >
            <svg
              aria-hidden="true"
              fill="none"
              focusable="false"
              height="32"
              viewBox="0 0 32 32"
              xmlns="http://www.w3.org/2000/svg"
            >
              
      
              <path
                d="M15.667 15.66v5.942H9.713V15.66zV9.722H4.807c-.58.004-1.047.47-1.043 1.04v15.755a1.04 1.04 0 0 0 1.028 1.054h15.771a1.04 1.04 0 0 0 1.042-1.04V15.66zm0-11.88v5.942h5.967v5.938h5.935V4.82a1.04 1.04 0 0 0-1.028-1.053h-.014z"
                fill="#6C6F77"
              />
              
      
            </svg>
            
    
          </span>
          <span
            aria-label="Confluence"
            class="css-1eu9lrx"
            role="img"
            style="--logo-color: inherit; --logo-fill: currentColor;"
          >
            <svg
              aria-hidden="true"
              fill="none"
              focusable="false"
              height="32"
              viewBox="0 0 32 32"
              xmlns="http://www.w3.org/2000/svg"
            >
              
      
              <path
                d="M25.769 20.234c-6.584-3.183-8.507-3.66-11.282-3.66-3.254 0-6.028 1.355-8.507 5.16l-.406.622c-.333.513-.407.696-.407.915s.11.403.517.659l4.18 2.598c.222.146.407.22.592.22.222 0 .37-.11.592-.44l.665-1.024c1.036-1.574 1.96-2.086 3.144-2.086 1.036 0 2.257.293 3.773 1.025l4.365 2.049c.443.22.924.11 1.146-.403l2.071-4.537c.222-.512.074-.842-.444-1.098M6.572 12.22c6.584 3.184 8.507 3.66 11.281 3.66 3.255 0 6.03-1.354 8.507-5.16l.407-.622c.333-.512.407-.695.407-.915s-.11-.402-.518-.658l-4.18-2.598c-.221-.147-.406-.22-.591-.22-.222 0-.37.11-.592.44l-.666 1.024c-1.035 1.573-1.96 2.086-3.144 2.086-1.035 0-2.256-.293-3.772-1.025L9.346 6.183c-.444-.22-.924-.11-1.146.402l-2.072 4.538c-.222.512-.074.841.444 1.098"
                fill="#6C6F77"
              />
              
    
            </svg>
          </span>
          <span
            aria-label="Jira"
            class="css-1eu9lrx"
            role="img"
            style="--logo-color: inherit; --logo-fill: currentColor;"
          >
            <svg
              aria-hidden="true"
              fill="none"
              focusable="false"
              height="32"
              viewBox="0 0 32 32"
              xmlns="http://www.w3.org/2000/svg"
            >
              
      
              <path
                d="M11.034 21.99h-2.22c-3.346 0-5.747-2.05-5.747-5.052h11.932c.619 0 1.019.44 1.019 1.062v12.007c-2.983 0-4.984-2.416-4.984-5.784zm5.893-5.967h-2.219c-3.347 0-5.748-2.013-5.748-5.015h11.933c.618 0 1.055.402 1.055 1.025V24.04c-2.983 0-5.02-2.416-5.02-5.784zm5.93-5.93h-2.219c-3.347 0-5.748-2.05-5.748-5.052h11.933c.618 0 1.018.439 1.018 1.025v12.007c-2.983 0-4.984-2.416-4.984-5.784z"
                fill="#6C6F77"
              />
              
    
            </svg>
          </span>
          <span
            aria-label="Jira Product Discovery"
            class="css-1eu9lrx"
            role="img"
            style="--logo-color: inherit; --logo-fill: currentColor;"
          >
            <svg
              aria-hidden="true"
              fill="none"
              focusable="false"
              height="32"
              viewBox="0 0 32 32"
              xmlns="http://www.w3.org/2000/svg"
            >
              
      
              <path
                d="M17.655 5.884a15.3 15.3 0 0 0 8.75 0c.257-.056.428.114.37.37-.826 2.985-.826 5.742 0 8.727.058.228-.113.427-.37.342a16.33 16.33 0 0 0-8.75 0c-.228.085-.428-.114-.342-.342.77-2.985.77-5.742 0-8.727-.086-.256.114-.426.342-.37m1.169 10.576v.483c0 2.246-1.568 3.894-3.506 3.894a3.474 3.474 0 0 1-3.477-3.496c0-1.933 1.567-3.468 3.762-3.468h.57V7.078c-.285-.028-.57-.028-.855-.028C9.618 7.05 5 11.655 5 17.37c0 5.713 4.617 10.29 10.318 10.29s10.346-4.605 10.346-10.29v-.91z"
                fill="#6C6F77"
              />
              
    
            </svg>
          </span>
          <span
            aria-label="Jira Service Management"
            class="css-1eu9lrx"
            role="img"
            style="--logo-color: inherit; --logo-fill: currentColor;"
          >
            <svg
              aria-hidden="true"
              fill="none"
              focusable="false"
              height="32"
              viewBox="0 0 32 32"
              xmlns="http://www.w3.org/2000/svg"
            >
              
      
              <path
                d="M18.79 13.257h7.325c1.1 0 1.476 1.046.805 1.878L15.464 29.274c-3.702-2.951-3.353-7.62-.644-11.027zm-5.66 5.634H5.806c-1.1 0-1.476-1.046-.805-1.878L16.457 2.874c3.702 2.951 3.3 7.566.617 11z"
                fill="#6C6F77"
              />
              
    
            </svg>
          </span>
          <span
            aria-label="Trello"
            class="css-1eu9lrx"
            role="img"
            style="--logo-color: inherit; --logo-fill: currentColor;"
          >
            <svg
              aria-hidden="true"
              fill="none"
              focusable="false"
              height="32"
              viewBox="0 0 32 32"
              xmlns="http://www.w3.org/2000/svg"
            >
              
      
              <path
                clip-rule="evenodd"
                d="M24.286 5.337H7.714a2.585 2.585 0 0 0-2.585 2.585v16.572a2.585 2.585 0 0 0 2.585 2.584h16.572a2.585 2.585 0 0 0 2.585-2.584V7.922a2.585 2.585 0 0 0-2.585-2.585m-9.791 15.66a.86.86 0 0 1-.859.858h-3.634a.86.86 0 0 1-.859-.859V10.21a.86.86 0 0 1 .859-.86h3.634a.86.86 0 0 1 .859.86zm8.362-4.953a.86.86 0 0 1-.859.859h-3.634a.86.86 0 0 1-.859-.859V10.21a.86.86 0 0 1 .859-.86h3.634a.86.86 0 0 1 .859.86z"
                fill="#6C6F77"
                fill-rule="evenodd"
              />
              
    
            </svg>
          </span>
        </div>
        <div
          role="presentation"
        >
          <button
            class="css-1ilq2uw"
            data-testid="login"
            type="button"
          >
            <span
              class="css-12rb8n9"
            >
              <span
                class="css-1gskvga"
                style="--logo-color: inherit; --logo-fill: currentColor;"
              >
                <svg
                  aria-hidden="true"
                  fill="none"
                  focusable="false"
                  height="32"
                  viewBox="0 0 32 32"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  
      
                  <path
                    d="M27.545 24.378 16.96 3.208c-.208-.458-.417-.541-.667-.541-.208 0-.458.083-.708.5-1.5 2.375-2.167 5.125-2.167 8 0 4.001 2.042 7.752 5.042 13.795.334.666.584.791 1.167.791h7.335c.541 0 .833-.208.833-.625 0-.208-.042-.333-.25-.75M12.168 14.377c-.834-1.25-1.084-1.334-1.292-1.334s-.333.083-.708.834L4.875 24.46c-.167.334-.208.459-.208.625 0 .334.291.667.916.667h7.46c.5 0 .875-.416 1.083-1.208.25-1 .334-1.876.334-2.917 0-2.917-1.292-5.751-2.292-7.251"
                    fill="inherit"
                  />
                  
    
                </svg>
              </span>
            </span>
            <span
              class="css-1nqby2s"
            >
              Login
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
